packages/tree-view/src/behaviors/TreeItemBehavior.js

```diff
@@ -57,9 +57,20 @@ const TreeItemBehavior = props => {
     }
   };
 
+  const handleDoubleClick = event => {
```

Comment:
Sorry I didn't think of this before, I haven't had a chance to pull this down, but have you tested if this works in both implementations:
- `TreeView` with `TreeItem` components as children
- `TreeView` using the `treeObject` prop

Reply:
The changes are made to TreeItem only, actually I did not notice treeObject before. So I tested the first case:
customized double click with event works well.
testing code:
```
        <TreeItem
          id="tree-item-0"
          key="tree-item-0"
          label="Tree Item 0"
          icon={<Hierarchy24 />}
          expandByDoubleClick="true"
          draggable="true"
          onDoubleClick={event => {
            // eslint-disable-next-line no-console
            console.log("*******", event.currentTarget);
          }}
        >
```
output:
<img width="1868" alt="Screen Shot 2022-01-27 at 9 52 48 PM" src="https://user-images.githubusercontent.com/37433962/151372323-a7180e6d-fd9d-4019-856d-40d3c70a8a8b.png">

For `treeObject`, it has nothing changed. Do you mean the enhancement should apply to  `treeObject` too? If so, I will add it later.

Reply:
Yes, the we want feature parity for people who use `TreeItem` as children or the `treeObject` prop.  Please apply the updated feature for that as well, as the `treeObject` is a little more optimized for larger data sets and should save on some renders, so I think more people may opt to use that prop instead.

Reply:
Make sense @wmui51.

Reply:
Applied the feature change to `treeObject`.

Tested with below codes:
- Customized double click action with `expandByDoubleClick`
-  Customized double click action only

```
const sampleTreeNodeObject = [
  {
    id: 1,
    parentId: null,
    meta: {
      label: "dbl click to expand/collapse",
      collapsed: false,
      active: false,
      expandByDoubleClick: true,
      icon: <Report24 />
    },
    draggable: true,
    onDoubleClick: event => {
      // eslint-disable-next-line no-console
      console.log("*******", event.currentTarget);
    },
    children: [
      {
        id: 2,
        parentId: 1,
        draggable: true,
        onDoubleClick: event => {
          // eslint-disable-next-line no-console
          console.log("*******", event.currentTarget);
        },
        meta: {
          label: "Tree Item 2",
          collapsed: false,
          active: false,
          icon: <Folder24 />
        },
...
```

Both works as expected:
<img width="1894" alt="Screen Shot 2022-01-28 at 3 19 37 PM" src="https://user-images.githubusercontent.com/37433962/151507165-be2fc0ba-6f35-496d-b383-b19f5526e10e.png">

@wmui51 Thanks for your continues reviewing and catching this.